Ability to edit comments

su7edja opened the pull request:
# What Changed
When making comment, --edit / -e would allow editing of existing comment. If no comment exists for a given context, a new comment is created.

# Why
fixes #516

Todo:

- [x] Add tests
- [x] Add docs
- [x] Add yourself to contributors (run `yarn contributors:add`)

<!-- GITHUB_RELEASE PR BODY: canary-version -->
Published PR with canary version: `7.7.0-canary.583.7506.0`
<!-- GITHUB_RELEASE PR BODY: canary-version -->

hipstersmoothie commented on packages/cli/src/parse-args.ts:
```diff
@@ -260,13 +267,15 @@ const commands: ICommand[] = [
       pr,
       context,
       { ...deleteFlag, description: 'Delete old comment' },
+      { ...editFlag, description: 'Edit old comment' },
```
Since the `--edit` flag is only used for the `comment` command can you move the full `editFlag` def here?

Similar to: https://github.com/intuit/auto/blob/ff1d187cb994e25f61655fb4ce8b5d93c47d2414/packages/cli/src/parse-args.ts#L303

su7edja replied:
@hipstersmoothie I moved the editFlag as you suggested.

For the linting errors, I fixed them all except the one in auto.ts. It's complaining about "no-collapsible-if  if statement can be merged with previous else" on lines 435 and 439.
But I'm unable to refactor it in a way such that the linter is happy and the code is still readable
Do you have any suggestions for this?

su7edja commented:
@hipstersmoothie I moved the editFlag as you suggested.

For the linting errors, I fixed them all except the one in auto.ts. It's complaining about "no-collapsible-if if statement can be merged with previous else" on lines 435 and 439.
But I'm unable to refactor it in a way such that the linter is happy and the code is still readable
Do you have any suggestions for this?

adierkens commented:
<!-- GITHUB_RELEASE COMMENT: released -->
:rocket: PR was released in v7.7.0 :rocket:
